### DISCUSSION TITLE
dynamically creating MST model based on a server response

### DISCUSSION THREAD

**Discussion by paxarpp** [mobxjs/mobx-state-tree#1729] [Q&A] [answered]
hi, is it possible to dynamically create a MST model, for example based on the server response, which will contain the structure of this model?

**Comment by cmdcolin** [comment]
seems like this should be possible, but it may be somewhat difficult to use this in practice e.g. how do you actually work with the model after you have created it? do you know ahead of time what keys you expect? then if so, maybe you are better off creating MST models up front statically

**Comment by paxarpp** [comment]
I need to change the values of the fields in the model, the types of fields and their names I want to take in the schema from the request to the server
```
example schema

[
  {
    type: number,
    name: 'count'
  },
...
]

I would like to receive model:

types.model(
  {
    count: types.number,
...
  }
)
```

**Reply by cmdcolin** [reply]
right, but given the fact that you already know what you expect to receive, why would you not statically create that model in your code?

**Comment by paxarpp** [comment]
I only know the format of the response for the schema, the name field can be any field name for the model.

**Comment by BATCOH** [comment, marked as answer]
Dynamic MST model creation is definitely possible with some boilerplate code.
However, this can be difficult to maintain and debug later on, especially if you want to use strong Typescript.

Quick and dirty basic example:
https://codesandbox.io/s/dynamic-mst-model-vyj4b?file=/src/index.js

**Comment by paxarpp** [comment]
thanks for the idea, I decided to stick with a types.custom
